File: TripKitAndroid/src/main/java/com/skedgo/tripkit/alerts/AlertBlock.java
package com.skedgo.tripkit.alerts;

import androidx.annotation.Nullable;

import com.google.gson.annotations.JsonAdapter;
import com.skedgo.tripkit.common.model.RealtimeAlert;

import static org.immutables.gson.Gson.TypeAdapters;
import static org.immutables.value.Value.Immutable;
import static org.immutables.value.Value.Style;

@Immutable
@TypeAdapters
@Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersAlertBlock.class)
public interface AlertBlock {
  @Nullable RealtimeAlert alert();

  @Nullable String disruptionType();
  @Nullable String[] operators();
  @Nullable Route[] routes();
  @Nullable ModeInfo modeInfo();
  @Nullable String[] stopCodes();
  @Nullable String[] serviceTripIDs();
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/alerts/AlertBlock.kt
package com.skedgo.tripkit.alerts

import com.google.gson.annotations.JsonAdapter
import com.skedgo.tripkit.common.model.realtimealert.RealtimeAlert
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@Immutable
@TypeAdapters
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersAlertBlock::class
)
interface AlertBlock {
    fun alert(): RealtimeAlert?

    fun disruptionType(): String?

    fun operators(): Array<String>

    fun routes(): Array<Route>

    fun modeInfo(): ModeInfo?

    fun stopCodes(): Array<String>

    fun serviceTripIDs(): Array<String>
}